r290 | kroeckx | 2008-03-23 18:52:37 +0100 (Sun, 23 Mar 2008) | 3 lines

Don't add extentions to ssl v3 connections.  It breaks with some
other software.  (Closes: #471681, #471896)

Index: openssl-0.9.8k/ssl/t1_lib.c
===================================================================
--- openssl-0.9.8k.orig/ssl/t1_lib.c	2008-09-04 00:13:04.000000000 +0200
+++ openssl-0.9.8k/ssl/t1_lib.c	2009-07-19 17:15:14.000000000 +0200
@@ -133,6 +133,10 @@
 	int extdatalen=0;
 	unsigned char *ret = p;
 
+	/* don't add extensions for SSLv3 */
+	if (s->client_version == SSL3_VERSION)
+		return p;
+
 	ret+=2;
 
 	if (ret>=limit) return NULL; /* this really never occurs, but ... */
@@ -251,6 +255,10 @@
 	int extdatalen=0;
 	unsigned char *ret = p;
 
+	/* don't add extensions for SSLv3 */
+	if (s->version == SSL3_VERSION)
+		return p;
+
 	ret+=2;
 	if (ret>=limit) return NULL; /* this really never occurs, but ... */
 
